src/state/shell/composer.tsx

```diff
@@ -51,11 +53,17 @@ export function Provider({children}: React.PropsWithChildren<{}>) {
         setState(opts)
       },
       closeComposer() {
-        setState(undefined)
+        let wasOpen = false
+        setState(v => {
+          wasOpen = !!v
+          return undefined
+        })
+        return wasOpen
```

Comment:
This will break in future versions of React, can we just return `state`?

Reply:
I think this can actually even break today too in some cases. React provides no guarantee at all that the updater function would run synchronously during the `setState` call

Reply:
Fixed